pygmt/tests/test_grdgradient.py

```diff
@@ -0,0 +1,36 @@
+"""
+Tests for grdgradient.
+"""
+import numpy.testing as npt
+import pytest
+from pygmt import grdgradient
+from pygmt.datasets import load_earth_relief
+from pygmt.exceptions import GMTInvalidInput
+
+
+@pytest.fixture(scope="module", name="grid")
+def fixture_grid():
+    """
+    Load the grid data from the sample earth_relief file.
+    """
+    return load_earth_relief(resolution="01d", region=[-5, 5, -5, 5])
+
+
+def test_grdgradient_no_outgrid(grid):
+    """
+    Test the below and above parameters for grdgradient with no set outgrid.
+    """
+    temp_grid = grdgradient(grid=grid, azimuth=10, direction="c")
+    assert temp_grid.dims == ("lat", "lon")
+    assert temp_grid.gmt.gtype == 1  # Geographic grid
+    assert temp_grid.gmt.registration == 1  # Pixel registration
+    npt.assert_allclose(temp_grid.min(), -0.0045060496)
+    npt.assert_allclose(temp_grid.max(), 0.0575332976)
+
+
```

Comment:
Could you add add an extra test using e.g. `grdgradient(..., radiance=[135, 45])` that produces an outgrid. Similar in style to your `test_grdclip_outgrid` example https://github.com/GenericMappingTools/pygmt/blob/ce774d4bcd5a0762991e80e1af3fe3c4819d091e/pygmt/tests/test_grdclip.py#L21-L35.

Reply:
For reasons I don't understand, both `test_grdgradient_outgrid()` and `test_grdgradient_no_outgrid()` fail when I run them on my computer, but they pass on the GitHub Action. For both of them, I get an `AssertionError` that tells me that the `assert_allclose` is not equal within tolerance. Any idea why this would be a problem when running the `test_grdgradient.py` on some machines by not others?

Reply:
Can you report the full error message? I'm not surprised that this did not work, this might be due to flakiness (#1242) and/or something with `grdgradient` itself. The `shading` parameter in `grdimage`/`grdview` (which uses `grdgradient` behind the scenes) has always been a bit sensitive to crashes and bugs, but we haven't managed to track down the true cause of it.

Reply:
Here is the failure for `test_grdgradient_outgrid()`:
![image](https://user-images.githubusercontent.com/29518865/118439771-87aa1e00-b6de-11eb-9077-d77b7f392850.png)

Reply:
Here is the failure for `test_grdgradient_no_outgrid()`:
![image](https://user-images.githubusercontent.com/29518865/118439855-a8727380-b6de-11eb-9e70-1d80feaeb460.png)

Reply:
@weiji14 What I'm a little confused about with the issue you mention above is that this test fails when I run `test_grdgradient.py` by itself and as part of `make test` on my computer , but consistently passes when the entire test suite is run on GitHub Actions. I would think the issue wouldn't affect a single file run and the issue is that all of the testing is run under one GMT instance.

Reply:
Could you provide the output of `pygmt.show_versions()` please? You're right that it's funny breaking the other way around. I'll try to see if I cam reproduce it on my end tomorrow.

Reply:
> PyGMT information:
  version: v0.3.2.dev143+g6bc7fa28
System information:
  python: 3.9.2 | packaged by conda-forge | (default, Feb 21 2021, 05:02:46)  [GCC 9.3.0]
  executable: /home/will/miniconda3/envs/gmt62rc/bin/python
  machine: Linux-5.8.0-53-generic-x86_64-with-glibc2.31
Dependency information:
  numpy: 1.20.2
  pandas: 1.2.4
  xarray: 0.17.0
  netCDF4: 1.5.6
  packaging: 20.9
  ghostscript: 9.53.3
  gmt: 6.2.0rc1
GMT library information:
  binary dir: /home/will/miniconda3/envs/gmt62rc/bin
  cores: 12
  grid layout: rows
  library path: /home/will/miniconda3/envs/gmt62rc/lib/libgmt.so
  padding: 2
  plugin dir: /home/will/miniconda3/envs/gmt62rc/lib/gmt/plugins
  share dir: /home/will/miniconda3/envs/gmt62rc/share/gmt
  version: 6.2.0rc1

I'm not too smart on using conda environments in PyCharm, but I changed the default conda environment to a different environment and changed it back, and now my test appear to be passing when both run alone and in `make test`. Not sure if it was a PyCharm error (unlikely) or human error (very likely), but the problem seems to be fixed.

Reply:
I can't seem to reproduce your error above when running `pytest  pygmt/tests/test_grdgradient.py`, so maybe you're right that there isn't any problem. My `pygmt.show_versions()` is as below for reference, just did a fresh install yesterday. I only see a difference in `xarray` and `ghostscript` versions but those shouldn't matter much I think.

```
PyGMT information:
  version: v0.3.2.dev145+ga2cdce2c.d20210520
System information:
  python: 3.9.4 | packaged by conda-forge | (default, May 10 2021, 22:13:33)  [GCC 9.3.0]
  executable: /home/username/miniconda3/envs/pygmt/bin/python
  machine: Linux-5.4.0-73-generic-x86_64-with-glibc2.31
Dependency information:
  numpy: 1.20.2
  pandas: 1.2.4
  xarray: 0.18.2
  netCDF4: 1.5.6
  packaging: 20.9
  ghostscript: 9.54.0
  gmt: 6.2.0rc1
GMT library information:
  binary dir: /home/username/miniconda3/envs/pygmt/bin
  cores: 6
  grid layout: rows
  library path: /home/username/miniconda3/envs/pygmt/lib/libgmt.so
  padding: 2
  plugin dir: /home/username/miniconda3/envs/pygmt/lib/gmt/plugins
  share dir: /home/username/miniconda3/envs/pygmt/share/gmt
  version: 6.2.0rc1
```